sharktank/sharktank/utils/testing.py

```diff
@@ -42,3 +43,84 @@ def run_main(self, main_func, *args):
     def assertFileWritten(self, p: Path):
         self.assertTrue(p.exists(), msg=f"Expected file {p} was not created")
         self.assertGreater(p.stat().st_size, 0, msg=f"Expected file {p} had zero size")
+
+
+def temporary_directory(identifier: str) -> tempfile.TemporaryDirectory:
+    """Returns a context manager TemporaryDirectory suitable for testing.
+
+    If the env var SHARKTANK_TEST_ASSETS_DIR is set then directories will be
+    created under there, prefixed by identifier and will not be deleted.
+
+    This is useful for getting updated goldens and such.
+    """
+    explicit_dir = os.getenv("SHARKTANK_TEST_ASSETS_DIR", None)
+    if explicit_dir is None:
+        return tempfile.TemporaryDirectory(prefix=f"{identifier}_")
+    else:
+        explicit_path = Path(explicit_dir)
+        explicit_path.mkdir(parents=True, exist_ok=True)
+        return tempfile.TemporaryDirectory(
+            prefix=f"{identifier}_", dir=explicit_path, delete=False
+        )
+
+
+@contextlib.contextmanager
+def override_debug_flags(flag_updates: dict):
+    from .debugging import flags
+
+    restore = {}
+    try:
+        for k, v in flag_updates.items():
+            print(f"Overriding debug flag {k} = {v}")
+            current_value = getattr(flags, k)
+            restore[k] = current_value
+            setattr(flags, k, v)
+        yield
+    finally:
+        for k, v in restore.items():
+            print(f"Restoring debug flag {k} = {v}")
+            setattr(flags, k, v)
+
+
+def get_best_torch_device() -> str:
+    import torch
+
+    if torch.cuda.is_available() and torch.cuda.device_count() > 0:
+        return "cuda:0"
+    return "cpu"
+
+
+def assert_golden_safetensors(actual_path, ref_path):
```

Comment:
This function is nice, giving a good reason to stay in Python instead of using bash or just the native tools like `iree-run-module`.

(I'm still slowly coming to terms with moving infrastructure from C/C++ to python :P)